circuit_maintenance_parser/data.py

```diff
@@ -23,16 +29,24 @@ def add_data_part(self, data_type: str, data_content: bytes):
         self.data_parts.append(DataPart(data_type, data_content))
 
     @classmethod
-    def init(cls, data_type: str, data_content: bytes):
+    def init_from_raw(cls: Type[T], data_type: str, data_content: bytes) -> Optional[T]:
```

Comment:
I'm not real familiar with this TypeVar pattern - would the following be as correct?

```suggestion
    def init_from_raw(cls: Type["NotificationData"], data_type: str, data_content: bytes) -> Optional["NotificationData"]:
```

Reply:
Doing like this, `mypy` complains with `Name 'Notification' is not defined  [name-defined]`
With `TypeVar` you create the actual Type for something that is unbound yet.
Maybe there is a better way to do...

Reply:
Huh. I really thought forward-declaration via strings was supported (are you sure you used `"NotificationData"` and not `"Notification"`?). Not blocking in any case though, was just curious.

Reply:
you were right!